### ISSUE TITLE
Unicode Completion in Search Bar

### ISSUE THREAD

**Issue by matthieugomez** [JuliaEditorSupport/Julia-sublime#106]
Unicode completion used to work in the search bar (the bar that pops up with CMD+F). It does not work after upgrading to ST4. Is that due to a change in Julia-sublime?

**Comment by randy3k** [comment]
We had our own completion API code in ST3 because the one in ST3 doesn't support backslashes properly. Those code were removed while upgrading to ST4 and the standard API for completion on ST4 is used now. Unfortunately, a standard API doesn't seem to support completion on search bar etc. We might need to re-enable the custom completion for search bar specifically.

**Comment by randy3k** [comment]
Fixed by 0f5fe40.

**Comment by matthieugomez** [comment]
Thanks a lot for solving this!
Btw, I love that \rho+ 1 tab automatically gives the right unicode. Is there a way to also have this behavior when typing in the editor? It's weird to have to do \rho + two tabs in sublime text v.s. \rho + one tab in the REPL

**Comment by randy3k** [comment]
`\rho` + one tab should give the right unicode. Are you on the master branch now?

**Comment by matthieugomez** [comment]
This would be really awesome! I think I'm on the master branch, but maybe
it's a problem with my installation[image: Screen Shot 2021-07-13 at
6.16.35 PM.png]

**Comment by matthieugomez** [comment]
This would be really awesome! I think I'm on the master branch, but maybe
there is a problem with my installation

<img width="209" alt="Screen Shot 2021-07-13 at 6 16 35 PM" src="https://user-images.githubusercontent.com/6223837/125644016-f3390203-295c-4256-9824-8fdc4b4f4e54.png">

**Comment by randy3k** [comment]
The pop up should appear automatically as you type, similar to what a normal completion works.

**Comment by matthieugomez** [comment]
Sorry, I'm a bit slow — I don't understand what you mean by "similar to how normal completion work" — I don't use any keyword completion in Sublime.

So let me be more precise about the current behavior on master (which is the same as in the last tagged version),
- when I type \rho nothing happens
- when i press tab, the pop appears (the one in the screenshot)
- when i press tab again (or enter), then \rho becomes ρ.
This is the same behavior as in the last tagged version.

In contrast, in Julia REPL, or the search field of sublime (thanks to your recent commit), \rho + tab directly becomes ρ

**Comment by randy3k** [comment]
> Maybe it's because I don't use any keyword completion in Sublime — it's the first time I'm seeing a popup like this

Then you must have disabled auto completion at all in the preference. How did you disable it?

**Comment by matthieugomez** [comment]
Thanks for helping me with this. I have this line in my preferences:
"auto_complete": false.
Indeed, if I set it to true, popup always appear, and \rho + tab gives ρ.

Documentation of Sublime Text 3 says:
"If auto complete is disabled, the completion popup can be shown manually, or tab can be used to insert the highest ranking completion without showing the popup."

Documentation of Subime Text 4 says:
"Automatically show the completions popup when typing."

So I guess the meaning of "auto-complete": false changed between the two versions. I am not sure how to disable the completetion popup while still allowing rho + tab to gives the best matching completion.

**Comment by matthieugomez** [comment]
loos like other people are angry about this ;) https://forum.sublimetext.com/t/st3-style-autocomplete-in-st4/57774/3

**Comment by matthieugomez** [comment]
Would there be a way to add the same behavior as before? I don't use tab autocompletion for anything except Julia unicode.

**Comment by matthieugomez** [comment]
Ok, so it seems that adding the following in key bindings did the trick
```julia
{
       "keys": ["tab"], "command": "julia_unicode_insert_best_completion", "context":
       [
        {"key": "selector", "operator": "equal", "operand": "source.julia"},
          { "key": "julia_unicode_has_matches"}
       ]
   },
```

**Comment by randy3k** [comment]
Oh ya, you got it. We have a manual command to insert best Unicode completion in ST3 and I have to bring it back to ST4 for the search bar. Although it is intended to be used in the search bar only, I'm glad that it also serves well for your purpose.

**Comment by randy3k** [comment]
Let me see how we should better support the users with `auto_complete` turned off.

**Comment by matthieugomez** [comment]
Thanks! Also, I pretty sure doing autocompletion (\rho+tab) used to work with multiple cursors in Sublime Text3, but it no longer works in Sublime Text 4.

**Comment by randy3k** [comment]
It's because you are using the `julia_unicode_insert_best_completion` that is designed to be used in the search bar. And of course multi curser is not a thing in search bar. It should be easy to extend to cover your case though.

**Comment by randy3k** [comment]
Fixed in 27e3acd. I actually was wrong. Multiple cursor is supported in the search bar.

**Comment by matthieugomez** [comment]
Thanks a lot!  Thank god for multiple cursors in the search bar 😅
